src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/PollingAppOfflineApplication.cpp

```diff
@@ -17,6 +17,8 @@ HRESULT PollingAppOfflineApplication::TryCreateHandler(_In_ IHttpContext* pHttpC
 void
 PollingAppOfflineApplication::CheckAppOffline()
 {
+    SRWSharedLock stopLock(m_stateLock);
```

Comment:
This would cause a deadlock. SRW is non-reentrant and non-upgradable Stop call might take a lock.

Reply:
This is also not required, `m_fStopCalled` is optimistic check it doesn't have to be correct every time.

Reply:
> non-reentrant

My bad, didn't realize that.